trlx/trainer/accelerate_base_trainer.py

```diff
@@ -86,8 +81,27 @@ def __init__(self, config, train_mode=True):
                 init_kwargs=init_trackers_kwargs,
             )
 
-        self.opt = self.setup_optimizer()
-        self.scheduler = self.setup_scheduler()
+    def setup_model(self):
```

Comment:
We didnt have a setup function before?

Reply:
Nope, we just created the model and froze parts of it in the constructor. Moving this into a setup function will easily let us add an optional model arg to the `__init__` so users can bring their own models in the future.

Reply:
Got it.